internal/generator/c/schema-reader.go

```diff
@@ -179,57 +185,114 @@ func (r *fbSchemaReader) readObjectField(entity *model.Entity, field *reflection
 	if strings.HasPrefix(comment, "objectbox:") || strings.HasPrefix(comment, "ObjectBox:") {
 		comment = strings.TrimSpace(comment[len("objectbox:"):])
 		if len(comment) == 0 {
 			return true, nil
 		}
-	} else {
-		return false, nil
+		return true, parseAnnotations(comment, annotations, supportedAnnotations)
 	}
+	return false, nil
+}
 
-	// sample content at this point:
-	// name="name",index
-	// id
+type annotInProgress struct {
+	name          string
+	key           string
+	value         *binding.Annotation
+	valueQuoted   bool
+	valueFinished bool
+}
 
-	type state struct {
-		name          string
-		value         *binding.Annotation
-		valueQuoted   bool
-		valueFinished bool
+func (s *annotInProgress) finishAnnotation(annotations *map[string]*binding.Annotation, supportedAnnotations map[string]bool) error {
+	s.name = strings.TrimSpace(s.name)
+	if len(s.name) == 0 {
+		return nil
+	}
+	if s.value == nil {
+		s.value = &binding.Annotation{} // empty value
+	} else {
+		s.value.Value = strings.TrimSpace(s.value.Value)
 	}
-	var s state
+	var key = s.key
+	if len(key) == 0 {
+		key = s.name
+	}
+	if (*annotations)[key] != nil {
+		return fmt.Errorf("duplicate annotation %s", key)
+	} else if !supportedAnnotations[s.name] {
+		return fmt.Errorf("unknown annotation %s", s.name)
+	} else {
+		(*annotations)[key] = s.value
+	}
+	return nil
+}
 
-	var finishAnnotation = func() error {
-		s.name = strings.TrimSpace(s.name)
-		if s.value == nil {
-			s.value = &binding.Annotation{} // empty value
-		} else {
-			s.value.Value = strings.TrimSpace(s.value.Value)
+// counts all "relation-" prefixed annotations (standalone relations) - used to ensure consistent processing order
+func relationsCount(annotations map[string]*binding.Annotation) uint {
+	var count uint
+	for key := range annotations {
+		if strings.HasPrefix(key, "relation-") {
+			count++
 		}
-		if (*annotations)[s.name] != nil {
-			return fmt.Errorf("duplicate annotation %s", s.name)
-		} else if !supportedAnnotations[s.name] {
-			return fmt.Errorf("unknown annotation %s", s.name)
-		} else {
-			(*annotations)[s.name] = s.value
-		}
-		s = state{}
-		return nil
 	}
+	return count
+}
 
-	for i, char := range comment {
-		if char == '=' && !s.valueQuoted { // start a value
+// parseAnnotations parses annotations in any of the following formats.
+// name="name",index - creates two annotations, name and index, the former having a non-empty value
+// relation(name=manyToManyRelName,to=TargetEntity) - creates a single annotation relation with two items as details
+// id - creates a single annotation
+// NOTE: this started as a very simple parser but it seems like the requirements are ever-increasing... maybe some form
+//       of recursive tokenization would be better in case we decided to rework.
+func parseAnnotations(str string, annotations *map[string]*binding.Annotation, supportedAnnotations map[string]bool) error {
```

Comment:
yes and I'm doing that but this line literally has 4 characters overhead and is perfectly fine to read because you know the complete line even if you can see the first 120 chars